crates/dock/tests/mutability_check.rs

```diff
@@ -0,0 +1,88 @@
+#[cfg(test)]
+mod tests {
+    use expect_test::expect_file;
+    use regex::Regex;
+
+    fn check(dir_name: &str) {
+        let mut out = Vec::new();
+        let mut err = Vec::new();
+
+        let main_file_contents =
+            std::fs::read_to_string(format!("tests/mutability_check/{dir_name}/main.nail"))
+                .unwrap();
+        // //---stdoutで区切ることで、標準出力と標準エラー出力をテストする
+        // //---stderrは必ず//---stdoutより後に書くことが前提
+        let (main_contents, _) = main_file_contents
+            .split_once("//---stdout")
+            .unwrap_or((&main_file_contents, ""));
+        let mut main_contents = main_contents.trim().to_string();
+
+        match dock::execute(
+            dock::NailExecutablePath::RootFile(
+                std::env::current_dir()
+                    .unwrap()
+                    .join(format!("tests/mutability_check/{dir_name}/main.nail")),
+            ),
+            &mut out,
+            &mut err,
+            false,
+        ) {
+            Ok(_) => (),
+            Err(e) => match e {
+                // 型チェックのエラーはExecutionError::Nailで返るため何もしない
+                // それ以外は型チェック以外のエラーなのでpanicしてテストを失敗させる
+                dock::ExecutionError::Nail => (),
+                dock::ExecutionError::Io(e) => panic!("io error: {e}"),
+                dock::ExecutionError::InvalidRootNailFilePath(e) => {
+                    panic!("invalid root nail file path: {e}")
+                }
+            },
+        }
```

Comment:
**Scalability Issue**: The execution function dock::execute lacks proper error handling for scalability. It uses panic for error handling, which is not suitable for a scalable system as it can terminate the entire process. <br> **Fix**: Replace panic with more resilient error handling mechanisms such as error logging and retries, or returning errors to the caller to handle them gracefully. <br> **Code Suggestion**: 
 ```
 -            Err(e) => match e {
 -                dock::ExecutionError::Nail => (),
 -                dock::ExecutionError::Io(e) => panic!("io error: {e}"),
 -                dock::ExecutionError::InvalidRootNailFilePath(e) => {
 -                    panic!("invalid root nail file path: {e}")
 -                }
 +            Err(e) => {
 +                log::error!("Execution error: {:?}", e);
 +                return Err(e);
 +            }
 ```